datadog/provider.go

```diff
@@ -35,6 +35,11 @@ func Provider() terraform.ResourceProvider {
 				Optional:    true,
 				DefaultFunc: schema.EnvDefaultFunc("DATADOG_HOST", nil),
 			},
+			"validate": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
```

Comment:
Yes, great idea!

Reply:
That would mean making making `api_key` and `app_key` not required though. With `validate = false` and no other changes you'd still need to set empty strings for the API and app keys.

Reply:
Ahh yeah good point! Maybe we could make api, app, and validate all optional, and just have quick validation in the provider init code that exactly one of these is set?

The expected workflow would then be:
Set the api/app key, and not validate. -> OK
Set validate and not api/app key -> OK
Set validate and api/app key -> Error message during provider initialization
Set neither -> Error message during initialization

Mostly I think it'd be helpful to avoid the scenario of a user not knowing which takes precedence; `validate` or the actual setting of the api/app keys.